bittensor/_axon/__init__.py

```diff
@@ -280,19 +274,19 @@ class AuthInterceptor(grpc.ServerInterceptor):
     def __init__(
         self,
         receiver_hotkey: str,
-        blacklist: Callable = None,
+        metagraph: "bittensor.metagraph",
     ):
         r"""Creates a new server interceptor that authenticates incoming messages from passed arguments.
         Args:
             receiver_hotkey(str):
                 the SS58 address of the hotkey which should be targeted by RPCs
-            black_list (Function, `optional`):
-                black list function that prevents certain pubkeys from sending messages
         """
         super().__init__()
         self.nonces = {}
-        self.blacklist = blacklist
+        self.blacklister = bittensor.auth_blacklister()
+        self.blacklist = self.blacklister.blacklist
         self.receiver_hotkey = receiver_hotkey
+        self.metagraph = metagraph
```

Comment:
Will this be the same metagraph object as the main metagraph in server? ie. when the metagraph gets synced in the server, will this metagraph also be update?

Reply:
Yes, it is the reference to the server metagraph from the axon. I will verify this, but am also having 2nd thoughts about including this if it means we make the AuthInterceptor heavier than need be.